src/strategies/discord.ts

```diff
@@ -0,0 +1,151 @@
+import { OAuth2Strategy, OAuth2StrategyVerifyCallback } from "./oauth2";
+
+export interface DiscordStrategyOptions {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  /**
+   * @default "identify email"
+   *
+   * See all the possible scopes:
+   * @see https://discord.com/developers/docs/topics/oauth2#shared-resources-oauth2-scopes
+   */
+  scope?: string;
+  prompt?: "none" | "consent";
+}
+
+/**
+ * @
+ */
+export interface DiscordProfile {
```

Comment:
To make it more consistent all OAuth2 based strategies should extend the OAuth2Profile interface which will define how to access certain common attributes like avatar or email, for any extra attribute not defined there you could do the same as the Auth9Profile and create an attribute with the whole object returned by Discord

Reply:
Sure, that makes total sense. I wasn't quite sure how to deal with it, but it's clear now.

I hope the way I dealt with the parameters photos and email fits. Since these are not always given depending on the scope or existing data.